#!/usr/bin/env python

PACKAGE = 'pmd_camboard_nano'

import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("remove_invalid_pixels", bool_t, 0, "Remove invalid pixels", True)

# Min and max values taken from "CamBoard Nano Plugins.pdf", page 4.
gen.add("integration_time", int_t, 0, "Integration time (us)", 333, 12, 2000)

# There does not seem to be a limitation on the maximum number of frames, yet
# there is not much sense to average over a long period.
gen.add("averaging_frames", int_t, 0, "Temporal sliding averaging window", 0,
        0, 120)

gen.add("bilateral_filter", bool_t, 0, "Bilateral filter", True)

exit(gen.generate(PACKAGE, "pmd_camboard_nano", "PMD"))
